core/document/src/layers/folder.rs

```diff
@@ -65,6 +65,84 @@ impl Folder {
 		Ok(())
 	}
 
+	pub fn reorder_layers(&mut self, source_ids: Vec<LayerId>, target_id: LayerId) -> Result<(), DocumentError> {
+		let source_pos = self.layer_ids.iter().position(|x| *x == source_ids[0]).ok_or(DocumentError::LayerNotFound)?;
+		let source_pos_end = source_pos + source_ids.len() - 1;
+		let target_pos = self.layer_ids.iter().position(|x| *x == target_id).ok_or(DocumentError::LayerNotFound)?;
+
+		let mut last_pos = source_pos;
+		for layer_id in &source_ids[1..source_ids.len()] {
+			let layer_pos = self.layer_ids.iter().position(|x| *x == *layer_id).ok_or(DocumentError::LayerNotFound)?;
+			if (layer_pos as i32 - last_pos as i32).abs() > 1 {
+				// Selection is not contiguous
+				return Err(DocumentError::InvalidPath);
```

Comment:
It's probably best to add a separate error code for that

Reply:
Cool, I wasn't sure but thought the same.